engine/context/src/main/java/io/deephaven/engine/context/QueryScope.java

```diff
@@ -119,6 +114,14 @@ private static Object applyValueConversions(final Object value) {
         return value;
     }
 
+    protected QueryScope() {
+        super(false);
+
+        // if (!Liveness.REFERENCE_TRACKING_DISABLED) {
```

Comment:
You're right (from slack chat) -- we should likely just check this inside of `incrementReferenceCount` which is a "public" API and used directly elsewhere (like directly by the `JobScheduler`). Let's get Ryan's opinion on this.

Reply:
To clarify a little bit:
`incrementReferenceCount()` is a public member of `ReferenceCounted`, but it's an implementation detail for `ReferenceCountedLivenessNode`. It's OK for an implementation to use internally if there's a good reason, but really `retainReference()` is the public API (and the implementation does check that flag).